### PR TITLE
Display time on modeline

### PR DESCRIPTION
**Pull request by jasonm23**
This is just for testing the install/functioning of ClockFace fonts in lambda line

### CODE REVIEW

File: lambda-line.el [hunk 1]
```diff
@@ -599,19 +599,105 @@ Otherwise show '-'."
       (concat (format-mode-line flymake-mode-line-format) " ")
     lambda-line--flycheck-text))
 
-;;;;; Display-time-mode
+;; Display-time-mode
+(defun lambda-line-install-clockface-fonts ()
```
Comment (jasonm23) [inline lambda-line.el:603]: Interactive command to install clockface font.

We can add this to README.md e.g.

> # Display time
>
> When Emacs `display-time-mode` is on, Lambda-line will show the time.  If you install [ClockFace Fonts](https://ocodo.github.io/ClockFace-font) the time can be displayed using those icon fonts.
>
> # Setup ClockFace fonts
>
> ```
> M-x lambda-line-install-clockface-fonts
> ```

(etc, there's a fair bit of material to cover, so we should finalise what's available / how before user docs.)

File: lambda-line.el [hunk 2]
```diff
@@ -599,19 +599,105 @@ Otherwise show '-'."
...
+  "Use a ClockFace font for unicode #xF0000..F008F.
+Optionally use a VARIANT."
+  (set-fontset-font
+    "fontset-default"
+    (cons (decode-char 'ucs #xF0000)
+          (decode-char 'ucs #xF008F))
+    (format "ClockFace%s" (or variant ""))))
+
+;; Usage example for testing
+;; - exal each one after font installation to test.
```
Comment (jasonm23) [inline lambda-line.el:672]: Give these user tests a try. @mclearc

Comment (mclearc) [reply lambda-line.el:672]: Great! I should be able to review all this later today.

**Comment by mclearc** [comment]
@jasonm23 this looks good. I think the icon needs at least one space (maybe 2) of left padding, as the percentage of buffer indicator seems to push right up against it when reading "top" or "bottom". I tested the install font function and the configure face function and both work well on macos (m1). 

I can add directions on installing the fonts and configuring to the docs. So I think once you're happy with the function and spacing we can merge.

**Comment by jasonm23** [comment]
I think the spacing is already in custom vars.  Is that incorrect?

**Comment by mclearc** [comment]
> I think the spacing is already in custom vars. Is that incorrect?

`lambda-line-time-icon-format` is currently set to "%s" and I was thinking it looks better with "  %s" as the default, given the buffer position size. 

The completing read selection is a nice touch.

**Comment by jasonm23** [comment]
Ok got you.  Seems fine on my display but I'm also using a variable pitch font on modeline so it's fairly atypical.

Happy to set defaults as you wish.

**Comment by mclearc** [comment]
Thanks again for your work!

**Comment by jasonm23** [comment]
https://github.com/Lambda-Emacs/lambda-line/blob/454289d62d8e44bc1e81abe23749256fe7c49280/lambda-line.el#L706

`lambda-line-icon-time` non nil stops displaying time textually.  Perhaps a better name is `lambda-line-icon-time-only`.  Either way, the README.org states that it turns on the icon. Which is always on if `display-time-mode` is on.

**Comment by jasonm23** [comment]
https://github.com/Lambda-Emacs/lambda-line/blob/454289d62d8e44bc1e81abe23749256fe7c49280/lambda-line.el#L710-L712 This is where the unicode char for the icon is inserted.  Fontset looks after selecting the font (defface / font-spec is ignored.)

**Pull request merged**